sktime/performance_metrics/forecasting/probabilistic/_classes.py

```diff
@@ -0,0 +1,177 @@
+        x_bar = self.evaluate(y_true, y_pred, multioutput, **kwargs)
+        for i in range(n):
+            out_series[i] = n * x_bar - (n - 1) * self.evaluate(
+                np.vstack((y_true[:i, :], y_true[i + 1 :, :])),
+                np.vstack((y_pred[:i, :], y_pred[i + 1 :, :])),
+                multioutput,
+            )
+        return out_series
+
+
+class PinballLoss(_BaseProbForecastingErrorMetric):
+    """Evaluate the pinball loss at the alpha quantile.
+
+    Parameters
+    ----------
+    alpha : The significance level required for the pinball loss
+
+    multioutput : string "uniform_average" or "raw_values" determines how multioutput
+    results will be treated
+    """
+
+    def __init__(self, alpha, multioutput="uniform_average"):
+        name = "PinballLoss"
+        # _tags = {"scitype:pred": "quantile"}
+        func = pinball_loss
+        self.alpha = alpha
+        super().__init__(func=func, name=name, multioutput=multioutput)
+
+    def evaluate(self, y_true, y_pred, multioutput="uniform_average", **kwargs):
+        """Evaluate the desired metric on given inputs."""
+        # Input checks
+        if isinstance(y_pred, pd.DataFrame):
+            if y_pred.columns.get_level_values(0)[0] == "Quantiles":
+                # Need to catch error if alpha not in dataframe
+                y_pred = y_pred.iloc[
+                    :, y_pred.columns.get_level_values(1) == self.alpha
+                ].to_numpy()
+
+        _, y_true, y_pred, multioutput = _check_reg_targets(y_true, y_pred, multioutput)
+        return self._evaluate(y_true, y_pred, multioutput)
+
+    def _evaluate(self, y_true, y_pred, multioutput):
+        # Return pinball loss
+        alpha = self.alpha
+        return pinball_loss(y_true, y_pred, alpha, multioutput)
+
+    def evaluate_by_index(self, y_true, y_pred, multioutput="uniform_average"):
+        """Return the metric evaluated at each time point."""
+        # Input checks
+        if isinstance(y_pred, pd.DataFrame):
+            if y_pred.columns.get_level_values(0)[0] == "Quantiles":
+                # Need to catch error if alpha not in dataframe
+                y_pred = y_pred.iloc[
+                    :, y_pred.columns.get_level_values(1) == self.alpha
+                ].to_numpy()
+
+        return super().evaluate_by_index(y_true, y_pred, multioutput)
+
+    @classmethod
+    def get_test_params(self):
+        """Retrieve test parameters."""
+        return {"alpha": 0.5}
+
+
+def pinball_loss(y_true, y_pred, alpha, multioutput):
+    """Evaluate the pinball loss at the alpha quantile.
+
+    Parameters
+    ----------
+    y_true : np.array of shape (fh,) or (fh, n_outputs) \
+             where fh is the forecasting horizon
+        Ground truth (correct) target values.
+
+    y_pred : np.array of shape (fh,) or (fh, n_outputs) \
+             where fh is the forecasting horizon
+        Forecasted alpha quantiles
+
+    alpha : The significance
```

Comment:
not a significance here, but the fraction at which the quantile is computed (not sure what the right term is, it's not the quantile itself)

Reply:
I think it would make sense to allow vectorization of alpha? Then do `np.array` casting?